### PR TITLE
style: update mat-card components

### LINKED ISSUE #258: Card components misaligned on Introduction page
# Bug Report

## Describe the Bug

After starting the application and navigating to the main introduction page, the card components do not display the content properly. Some padding is missing, causing the layout to appear distorted.

### Expected Behavior

Card elements need to properly display their inner content.

### Observed Behavior

Some padding is missing, which causes the layout to look distorted.

## Steps to Reproduce

1. Clone the repository.
2. Navigate to the project directory and run the command: **npm install**.
3. Start the application by running the command: **npm start**.

## Context Information

- OS: [macOS]
- Web Browser: [Brave]

## Detailed Description

localhost_4200_introduction(lap) (1)

## Possible Implementation

Add padding to enhance the user experience.

### PR DESCRIPTION
**Pull request by s-damith**
## What this PR changes/adds

Refactor Angular Material mat-card components to enhance UI and functionality on the introduction page.

## Why it does that

Add padding to the internal elements of the card components and utilize standardized card component sub-elements.

localhost_4200_introduction(lap) (2)

## Linked Issue(s)

Closes #258 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: src/modules/edc-demo/components/introduction/introduction.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div id="wrapper">
     <mat-card>
-    <h1>Eclipse Dataspace Connector</h1>
-    <div>
+      <mat-card-header><h1>Eclipse Dataspace Connector</h1></mat-card-header>
+    <mat-card-content>
```
Comment (timdah) [inline src/modules/edc-demo/components/introduction/introduction.component.html:4]:
```suggestion
      <mat-card-header>
        <mat-card-title>Eclipse Dataspace Connector</mat-card-title>
      </mat-card-header>
      <mat-card-content>
```

Using `mat-card-title` to follow material design.

File: src/modules/edc-demo/components/introduction/introduction.component.html [hunk 2]
```diff
@@ -26,12 +26,12 @@ <h1>Eclipse Dataspace Connector</h1>
...
         </p>
-    </div>
+    </mat-card-content>
 </mat-card>
 
 <mat-card>
-    <h1>EDC Data Dashboard (this application)</h1>
-    <div>
+  <mat-card-header><h1>EDC Data Dashboard (this application)</h1></mat-card-header>
+    <mat-card-content>
```
Comment (timdah) [inline src/modules/edc-demo/components/introduction/introduction.component.html:34]:
```suggestion
    <mat-card-header>
      <mat-card-title>EDC Data Dashboard (this application)</mat-card-title>
    </mat-card-header>
    <mat-card-content>
```

File: src/modules/edc-demo/components/introduction/introduction.component.scss [hunk 3]
```diff
@@ -14,7 +14,6 @@ h1 {
 
 .description {
     text-align: justify;
-    padding-left: 20px;
     padding-right: 20px;
```
Comment (timdah) [inline src/modules/edc-demo/components/introduction/introduction.component.scss:17]: Remove padding-right as well for symmetrical look.

**Review by timdah — CHANGES_REQUESTED** [review]
I would also propose to set the font of mat-card (`font-family: 'Open Sans'`) for a uniform typeface.

**Review by timdah — CHANGES_REQUESTED** [review]
Added the suggested font-family. The rest looks good to me.

File: src/modules/edc-demo/components/introduction/introduction.component.scss [hunk 4]
```diff
@@ -14,8 +14,6 @@ h1 {
 
 .description {
     text-align: justify;
-    padding-left: 20px;
-    padding-right: 20px;
 }
 
 mat-card {
```
[the remark this answers is not in the document]
Comment (s-damith) [reply src/modules/edc-demo/components/introduction/introduction.component.scss:19]: Hi! Replace the `<h1>` tag with `mat-card-title`. It applies the 'Open Sans' font family by default, so there's no need to apply the font family again.

Screenshot 2024-12-19 at 4 35 07 PM

[the remark this answers is not in the document]
Comment (s-damith) [reply src/modules/edc-demo/components/introduction/introduction.component.scss:19]: Yeah, you are right. But the description text uses a different font type. Therefore, I added the font family to mat-card. Thanks.

[the remark this answers is not in the document]
Comment (timdah) [reply src/modules/edc-demo/components/introduction/introduction.component.scss:19]: Hi! True, but the paragraphs inside `mat-card-content` still have the wrong font. For the sake of simplicity, I would therefore set it for the whole card.

**Review by timdah — APPROVED** [review]

**Pull request merged**